### PR TITLE
Endpoint for getting user details.

### PR DESCRIPTION
**Pull request by webnard**

### CODE REVIEW

File: hummedia/resources.py [hunk 1]
```diff
@@ -103,6 +108,42 @@ def set_query(self):
...
+        from mongokit import cursor
+        from bson import ObjectId
+        from models import connection
+        from config import APIHOST, YT_SERVICE, BYU_WS_ID, BYU_SHARED_SECRET
+        from urllib2 import Request, urlopen, URLError
+        import json, byu_ws_sdk, requests, re, os, mimetypes
+        import time
+        import xml.etree.ElementTree as ET
+
+        def getCurrentSem():
```
Comment (dcravenus) [inline hummedia/resources.py]: Is this function always accurate?  Doesn't spring/summer share the month of June or something?

File: hummedia/resources.py [hunk 2]
```diff
@@ -103,6 +108,42 @@ def set_query(self):
...
+            elif today.month in [9,10,11,12]:
+                sem="5"
+            return str(today.year)+sem
+
+        url="https://ws.byu.edu/rest/v1.0/academic/registration/studentschedule/"+userid+"/"+getCurrentSem()
+        headerVal = byu_ws_sdk.get_http_authorization_header(BYU_WS_ID, BYU_SHARED_SECRET, byu_ws_sdk.KEY_TYPE_API,byu_ws_sdk.ENCODING_NONCE,actor=username,url=url,httpMethod=byu_ws_sdk.HTTP_METHOD_GET,actorInHash=True)
+        res = requests.get(url, headers={'Authorization': headerVal})
+        try:
+            return json.loads(res.text)
+        except Exception:
```
Comment (dcravenus) [inline hummedia/resources.py]: Do we know the specific exception we are trying to catch?

File: hummedia/resources.py [hunk 3]
```diff
@@ -103,6 +108,42 @@ def set_query(self):
         else:
             q["oauth"][self.manual_request["provider"]]=self.manual_request["provider_id"]
         return q
+
+    def get_enrollments(self, username, userid):
+        from flask import request, Response, jsonify, current_app, session
+        from flask_oauth import OAuth
+        from datetime import datetime, timedelta, date
+        from functools import update_wrapper
+        from mongokit import cursor
```
Comment (dcravenus) [inline hummedia/resources.py]: Are we using all these imports?

Comment (webnard) [inline hummedia/resources.py]: image

File: hummedia/resources.py (hunk 1 above)
Comment (webnard) [inline hummedia/resources.py]: You're right. There seems to be an updated getCurrentSem in the helpers.py file. I'll import it.

**Comment by webnard** [comment]
@dcravenus I think I've fixed these issues if you don't mind taking another look. Thanks

**Comment by dcravenus** [comment]
image
Looks good

**Pull request merged**